src/openassetio-core/include/openassetio/managerApi/ManagerInterface.hpp

```diff
@@ -681,6 +681,21 @@ class OPENASSETIO_CORE_EXPORT ManagerInterface {
                        const ResolveErrorCallback& errorCallback) = 0;
 
   /// @}
+ protected:
+  /**
+   * Create an @ref EntityReference object wrapping a given @ref
+   * entity_reference string. This should be used for all reference
+   * creation by a manager's implementation.
+   *
+   * No validation is performed as this method is only visible to the
+   * manager implementation, and so it is assumed that its internal
+   * business logic inherently ensures only valid strings are
+   * returned.
+   *
+   * @param entityReferenceString Raw string representation of the
+   * entity reference.
+   */
+  [[nodiscard]] EntityReference createEntityReference(Str entityReferenceString) const;
```

Comment:
[consider] I think this should be public and used by the `Manager`'s `createEntityReference*` functions. DRY and whatnot.  

Assuming we eventually bundle a reference/ID to the manager implementation in the entity reference, for validation purposes, this then gives us a handy factory function to change.

Reply:
My concern here, is that this method deliberately _doesn't_ validate the string, as it is assumed that it is a valid ref as it is being supplied by the managers implementation (and can only be). If we make this public, that is no longer assured, which then makes it feel like this method should validate, which is the thing its deliberately avoiding.

Reply:
Capturing offline discussion, we'll defer this conversation until we look into enhanced validation of entity references. Making this `protected` for now and potentially making `public` later is safer than the reverse.